framework/src/meshmodifiers/MeshModifier.C

```diff
@@ -19,6 +19,9 @@ InputParameters validParams<MeshModifier>()
 {
   InputParameters params = validParams<MooseObject>();
   params.addParam<std::vector<std::string> >("depends_on", "The MeshModifiers that this modifier relies upon (i.e. must execute before this one)");
+  params.addParam<bool>("force_prepare", false,
```

Comment:
It does need to be user settable. You only need to force a prepare if you are running subsequent modifiers that need information from the current modifier (this is a rare case).